sympy/polys/matrices/domainmatrix.py

```diff
@@ -184,18 +557,90 @@ def inv(self):
         return self.from_rep(inv)
 
     def det(self):
+        r"""
+            Returns the determinant value for a Square DomainMatrix
+
+            Usage
+            =====
+
+            >>> from sympy import Matrix
+            >>> from sympy.polys.matrices import DomainMatrix
+            >>> Matrix1 = Matrix([
+            ...         [2, 73],
+            ...         [27, 86]])
+            >>> domainMatrix = DomainMatrix.from_Matrix(Matrix1)
+            >>> domainMatrix
+            DomainMatrix([[2, 73], [27, 86]], (2, 2), ZZ)
+
+            >>> domainMatrix.det()
+            -1799
```

Comment:
It's nicer to pick examples that someone can do with mental arithmetic so that they can quickly see that the result is what they expected.